File: SortingAndSearching/FerrisWheel.java
import java.util.Arrays;
import java.util.Scanner;

public class FerrisWheel {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int x = sc.nextInt();
        int[] a = new int[n];
        for(int i = 0 ; i < n; i++){
            a[i] = sc.nextInt();
        }
        Arrays.sort(a);
        int count = 0;
        int i = 0, j = n-1;
        while(i <= j){
            if(a[i] + a[j] <= x){
                count++;
                i++;
                j--;
            }else{
                j--;
                count++;
            }
        }
        System.out.println(count);


    }
}

File: SortingAndSearching/FerrisWheel.cpp
#include <iostream>
#include <bits/stdc++.h>
using namespace std;
int main(){
    int n, x;
    cin >> n >> x;
    int a[n];
    for(int i = 0 ; i < n; i++){
        cin >> a[i];
    }
    sort(a, a+n);

    int count = 0;
    int i = 0, j = n-1;
    while(i <= j){
        if(a[i] + a[j] <= x){
            count++;
            i++;
            j--;
        }else{
            j--;
            count++;
        }
    }
    cout << count;
}